packages/dds/tree/src/test/README.md

```diff
@@ -2,3 +2,35 @@
 
 This folder contains tests for the whole package.
 The folder hierarchy within mirrors that of the rest of the `src` directory.
+
+In general, these tests should be organized such that every file under `src` has a corresponding file under `src/test`.
+
+Files under `src/test/` are identified by their "test file kind", which is indicated by an extra `.` delimited section before the file extension. For example `foo.spec.ts` is a `spec` test file.
+
+Files `src/test/` should fall into one of these kinds:
+
+ - None: These are regular TypeScript files, like `foo.ts`. These should never contain top level test suites. They are files which exist to be imported by other kinds of test files. They can contain functions which generate test suites, or other test utilities.
+ - `spec`: These are Mocha test suites, and should correspond to a matching file under `src`.
+For example `src/foo/bar.ts` would have its test suite in `src/test/foo/bar.spec.ts`. These files should define a single top level suite (using `describe`) which is named after the file being tested (without extension).
+If ambiguous, additional qualifiers can be added to the suite name, like "simple-tree util" if this name would collide.
+Unless otherwise documented in that file, this file should contain the unit tests for the logic in the corresponding source file.
+Unless otherwise documented, the tests should be organized matching the structure of the file being tested.
+For example every top level item in the file gets its own test or test suite inside the field/module's test suite, and this is recursively applied to the inner members/suites.
+If test coverage of the logic is tested elsewhere (for example as part of some other nested suite in the same file, or as part of some other file's tests), this should be noted with a comment in both locations.
+These `spec` tests should confirm that each member in the file being tested meets its documented specification (its API's doc comments, and any additional behavior implied by its types or naming): again exceptions to this are allowed but should be documented.
+Assuming all code has such tests, and only relies of documented behavior of the code its using, tests following this pattern should be sufficient for localizing bugs (determining which code fails to conform to its documentation, regardless of if its due to depending on undocumented behaviors of other code or some other kind of logic bug).
+In practice testing of this type will not cover all cases, and this is fine, but when a testing gap is found and determined significant (for example due to a bug, new code, or audit of coverage), it should always be possible to address it by adding more tests following this pattern (for example regression tests can always be implemented as a `spec` test for the code which contained the bug).
```

Comment:
Nit: this is a bit of a run-on. Probably worth some restructuring.

Reply:
Should be better now.